src/prof_sys.c

```diff
@@ -22,9 +22,15 @@
 #define _Unwind_Backtrace JEMALLOC_TEST_HOOK(_Unwind_Backtrace, test_hooks_libc_hook)
 #endif
 
+#ifdef JEMALLOC_PID_NAMESPACE
+#include "jemalloc/internal/pid_namespace_externs.h"
+#endif
+
+
 /******************************************************************************/
 
 malloc_mutex_t prof_dump_filename_mtx;
+malloc_mutex_t prof_pid_namespace_mtx;
```

Comment:
No need for the mutex since opt_prof_pid_namespace is not allowed to change (unlike the filename prefix)